src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/EmailNotificationService.java

```diff
@@ -0,0 +1,43 @@
+package uk.gov.hmcts.ethos.replacement.docmosis.service;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.ecm.common.idam.models.UserDetails;
+import uk.gov.hmcts.et.common.model.ccd.CaseUserAssignment;
+import uk.gov.hmcts.ethos.replacement.docmosis.domain.ClaimantSolicitorRole;
+import uk.gov.hmcts.ethos.replacement.docmosis.domain.SolicitorRole;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class EmailNotificationService {
+    private final AdminUserService adminUserService;
+
+    public List<String> getCaseClaimantSolicitorEmails(List<CaseUserAssignment> assignments) {
+        List<String> emailAddresses = new ArrayList<>();
+        for (CaseUserAssignment assignment : assignments) {
+            if (ClaimantSolicitorRole.CLAIMANTSOLICITOR.getCaseRoleLabel().equals(assignment.getCaseRole())) {
+                UserDetails userDetails = adminUserService.getUserDetails(adminUserService.getAdminUserToken(),
+                        assignment.getUserId());
+                emailAddresses.add(userDetails.getEmail());
+            }
+        }
+        return emailAddresses;
+    }
+
+    public List<String> getRespondentSolicitorEmails(List<CaseUserAssignment> assignments) {
```

Comment:
Rather than having a List, should probably change this to a Set just to handle duplicate values. if a solicitor is SOLICITORA and SOLICITORB, current solution will store the emails twice and will send out 2 of the same emails. Using a Set prevents duplicates so even if they are representing more than 1 respondent, they only get 1 email